test/cases/db-update.bats

```diff
@@ -94,6 +94,33 @@ load ../lib/common
 	[[ -f ${ARCHIVE_BASE}/packages/p/pkg-any-a/pkg-any-a-1-1-any${PKGEXT}.sig ]]
 }
 
+@test "update any package to stable repo without updating testing package fails" {
+	releasePackage extra pkg-any-a
+	db-update
+	updatePackage pkg-any-a
+	releasePackage testing pkg-any-a
+	db-update
+	updatePackage pkg-any-a
+	releasePackage extra pkg-any-a
+
+	run db-update
+	[ "$status" -ne 0 ]
+}
+
+@test "update any package to stable repo without updating staging package fails" {
+	releasePackage extra pkg-any-a
+	db-update
+	updatePackage pkg-any-a
+	releasePackage staging pkg-any-a
+	db-update
+	updatePackage pkg-any-a
+	releasePackage extra pkg-any-a
+
+	run db-update
+	echo "$output"
```

Comment:
Is this  echo intentional or is that left over from debugging?

Reply:
It's arguably a good thing. If the following test for the return code fails, the output of the command silently disappears.

bash-bats `run` is basically a low-level tool to reimplement xfail. It's annoying that it doesn't echo output when a test errors due to the test command returning true -- it makes it difficult to follow what happened, and I hit that when looking at this testcase. Maybe I should change this everywhere...